start_admin.sh

```diff
@@ -181,8 +181,19 @@ if kex_algorithms=$(jq -r -e -c '.["ssh"]["key-exchange-algorithms"]? | join(","
 fi
 
 # Populate MACs with all the MACs found in user-data
-if macs=$(jq -r -e -c '.["ssh"]["macs"]? | join(",")' "${USER_DATA}" 2>/dev/null); then
-  echo "MACs ${macs}" >> "${SSHD_CONFIG_FILE}"
+  if macs=$(jq -r -e -c '.["ssh"]["macs"]? | join(",")' "${USER_DATA}" 2>/dev/null); then
+    echo "MACs ${macs}" >> "${SSHD_CONFIG_FILE}"
+  fi
+
+# Set custom SSH port if specified in user-data
+if ssh_port=$(jq -r -e '.["ssh"]["port"]?' "${USER_DATA}" 2>/dev/null); then
+  # Validate port number (must be between 1 and 65535)
+  if [[ "${ssh_port}" =~ ^[0-9]+$ ]] && [ "${ssh_port}" -ge 1 ] && [ "${ssh_port}" -le 65535 ]; then
```

Comment:
```suggestion
  if [[ ${ssh_port} -ge 1 ]] && [[ ${ssh_port} -le 65535 ]]; then
```
This can be simplified a bit. Also no need to quote ints. The ssh server will do its own validation so you could even just do `[[ ${ssh_port} -ge 1 ]]`

Reply:
Removed conditionals in revision